src/main/java/com/github/nylle/javafixture/InstanceFactory.java

```diff
@@ -0,0 +1,72 @@
+package com.github.nylle.javafixture;
+
+import org.objenesis.Objenesis;
+import org.objenesis.ObjenesisStd;
+import org.objenesis.instantiator.ObjectInstantiator;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Proxy;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import static java.lang.String.format;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toList;
+
+public class InstanceFactory {
+
+    private final SpecimenFactory specimenFactory;
+    private final Random random;
+
+    public InstanceFactory(SpecimenFactory specimenFactory) {
+        this.specimenFactory = specimenFactory;
+        this.random = new Random();
+    }
+
+    public <T> T construct(final SpecimenType<T> type) {
+        var constructors = stream(type.asClass().getConstructors())
+                .filter(x -> Modifier.isPublic(x.getModifiers()))
```

Comment:
I  think only public constructors  make sense because I want an object that *could* have been created in my program. Maybe, just maybe we could also allow protected.

Reply:
protected could be used from within the same package, so we could allow these too

Reply:
You sure? I think `protected` only works from a derived class. What you mean is package-protected (no modifier).